### DISCUSSION TITLE
TSkAnimatedImage to other images types

### DISCUSSION THREAD

**Discussion by cayque10** [skia4delphi/skia4delphi#243] [General]
Hello, guys.

I have a question about the TSkAnimatedImage. It's possible de conversation from TSkAnimatedImage to other images types like PNG, Bitmap, JPG? 

I know the internal structure of the data is quite different, but is there any way to do this?

I'm tryed use the TMemoryStream to read the **TSkAnimatedImage.Source.Data** , but I received an error when save to other images types.

Thank you.

**Comment by viniciusfbb** [comment]
Hi @cayque10!

Currently, the TSkAnimatedImage accept the types:
 - Animated WebP (.webp)
 - Animated GIF (.gif)
 - Lottie animation (.lottie or .json)
 - Telegram sticker (.tgs)

The TSkAnimatedImage.Source.Data is the TBytes of the original data that you has loaded. It doesn't do any kind of conversation, that is, if you load a lottie animation, that bytes will be a json bytes of the lottie animation. If you load a GIF image, that bytes will be the binary of the original GIF image.

If you want a screenshot, if you are on FMX, you can do a simple TSkAnimatedImage.MakeScreenshot to get the TBitmap screenshot. (Remember that in this case, you should call free on the returned bitmap by you self).

**Comment by cayque10** [comment]
Hello, @viniciusfbb . Thank you for the feedback. 
I used the PaintTo method to get a print. 
Thank you.
